### PR TITLE
multi-env support

### LINKED ISSUE #36: Apigee Environment multi-tenancy
Quota:
- Apigee environment needs to be put into the quota identifiers. This is a breaking change similar to #34 

Products:
- `APIProduct` contains a map of Apigee environment instead of a list for efficient lookups.
- Apigee environment in `auth.Context` will be used to match products or operation configs.

### PR DESCRIPTION
**Pull request by rockspore**
* using auth.Context.Environment() for authorizing
* quota identifier now includes the environment
* authorize tests are refactored with tables of test cases
fix #36

### CODE REVIEW

**Comment by rockspore** [comment]
Converted to draft to make changes regarding https://github.com/apigee/apigee-remote-service-cli/issues/144

Also will double check the existing code change.

**Comment by theganyo** [comment]
@rockspore Can you take a look at my changes?

File: analytics/manager.go [hunk 1]
```diff
@@ -260,18 +260,13 @@ func (m *manager) startUploader(errHandler util.ErrorFunc) {
...
 		if err == nil {
-			prometheusRecordsByFile.Delete(promLabels)
-			org, env, err := getOrgAndEnvFromTenant(tenant)
-			if err != nil {
-				log.Warnf("unable to set uploaded status count: %s", err)
-			} else {
-				countLabels := prometheus.Labels{"org": org, "env": env, "status": "uploaded"}
-				prometheusRecordsCount.With(countLabels).Add(float64(numRecs))
-			}
+			org, env, _ := getOrgAndEnvFromTenant(tenant)
```
Comment (rockspore) [inline analytics/manager.go:266]: Why are we not checking the error from this method any more?

Comment (theganyo) [reply analytics/manager.go:266]: There's no point. 1. It should never fail anyway, but also: 2. If it fails, this way we can still track in prometheus.

**Comment by rockspore** [comment]
> @rockspore Can you take a look at my changes?

LGTM. You are still the reviewer so please pull the trigger :)

**Review by theganyo — APPROVED** [review]

**Pull request merged**